crates/egui/src/text_selection/text_cursor_state.rs

```diff
@@ -302,18 +302,108 @@ pub fn find_line_start(text: &str, current_index: CCursor) -> CCursor {
     // number of multi byte chars.
     // We need to know the char index to be able to correctly set the cursor
     // later.
-    let chars_count = text.chars().count();
+    let mut char_line_indexes: Vec<usize> = create_char_line_indexes(text);
+    let line_start_index = get_line_start_index(&mut char_line_indexes, current_index.index);
 
-    let position = text
-        .chars()
-        .rev()
-        .skip(chars_count - current_index.index)
-        .position(|x| x == '\n');
+    CCursor::new(line_start_index)
+}
+
+pub fn create_char_line_indexes(text: &str) -> Vec<usize> {
+    let mut line_indexes = vec![0];
+
+    let mut count = 0;
+    for c in text.chars() {
+        count += 1;
+        if c == '\n' {
+            line_indexes.push(count);
+        }
+    }
+    line_indexes.push(count);
+
+    line_indexes
+}
+
+pub fn get_line_start_index(line_indexes: &mut [usize], index: usize) -> usize {
+    let (_current_line, line_start) = get_line_and_start_index(line_indexes, index);
+    line_start
+}
+
+pub fn get_line_and_start_index(line_indexes: &mut [usize], index: usize) -> (usize, usize) {
+    let mut current_line = 1;
+    for (i, line_start) in line_indexes.iter().enumerate() {
+        current_line = i;
+        if *line_start > index {
+            break;
+        }
+    }
+
+    (current_line, line_indexes[current_line - 1])
+}
 
-    match position {
-        Some(pos) => CCursor::new(current_index.index - pos),
-        None => CCursor::new(0),
+pub fn get_current_column(text: &str, line_indexes: &mut [usize], index: usize) -> usize {
+    let (current_column, _total_column) =
+        get_current_column_total_column(text, line_indexes, index);
+    current_column
+}
+
+pub fn get_current_column_total_column(
```

Comment:
you shouldn't be using `get_` everywhere

https://rust-lang.github.io/api-guidelines/naming.html#getter-names-follow-rust-convention-c-getter

Reply:
> you shouldn't be using `get_` everywhere
> 
> https://rust-lang.github.io/api-guidelines/naming.html#getter-names-follow-rust-convention-c-getter

I'm not from an English-speaking country, so it's hard for me to choose even a single word.
I would appreciate it if `emilk` could make a suggestion.